lib/proxy/sd_notify.rb

```diff
@@ -12,12 +18,45 @@ def notify(message)
         socket.sendmsg(message.chomp + "\n") # ensure trailing \n
         socket.close
       end
+    rescue Exception => e
+      raise "NOTIFY_SOCKET was set but unable to open: #{e}"
```

Comment:
Isn't this misleading? It can also be that it failed to send on the socket or otherwise. Don't you lose the context of the original exception this way?

Reply:
Reworded, true it's sub ideal. Not sure what you mean by loosing a context - this is re-raise that goes up the stack, not a handling code.